dnf-behave-tests/dnf/steps/fixtures/httpd.py

```diff
@@ -51,7 +51,14 @@ def do_GET(self):
             self.send_response(self.server._conf['status'])
             self.end_headers()
             return
-        super(LoggingHttpHandler, self).do_GET()
+        if self.server._conf.get("response_url"):
+            if self.server._conf["response_url"] == "/{context.dnf.tempdir}/coprs/testuser/testproject/repo/fedora-31" \
+                                                    "/dnf.repo":
+                self.send_response(404)
```

Comment:
For that matter, this also isn't good. The particular response status code should also in general be stored in the same per-path dict. The global status that is being returned above (on line 51) is a simplification because we somehow never needed that to be different for two different URLs in a single scenario (the server is mostly used to serve files from a directory).

I think you should be fine with that single global status, so just put your code before that and then set the global status to be 404 in your step in the scenario, so that there's no specific 404 response in case you have extra headers configured, because those are unrelated.